plugins/linguist-backend/api-report.md

```diff
@@ -49,11 +37,15 @@ export class LinguistBackendDatabase implements LinguistBackendStore {
   // (undocumented)
   static create(knex: Knex): Promise<LinguistBackendStore>;
   // (undocumented)
+  deleteEntity(entityRef: string): Promise<void>;
+  // (undocumented)
+  getAllEntities(): Promise<string[]>;
+  // (undocumented)
   getEntityResults(entityRef: string): Promise<Languages>;
   // (undocumented)
-  getProcessedEntities(): Promise<ProcessedEntity[] | []>;
+  getProcessedEntities(): Promise<ProcessedEntity[]>;
   // (undocumented)
-  getUnprocessedEntities(): Promise<string[] | []>;
+  getUnprocessedEntities(): Promise<string[]>;
```

Comment:
Do these really need to be implemented by everyone? It looks like a lot of these are for internal workings and testings.

I would try to slim down the interface to only what's needed for the backend to function and construct as much as possibly inside the router(privately) unless there really need to be possibly to supply alternatives.
Your default client that's not exported could have all of these "public" but internal for you to properly test if that makes sense :)

Reply:
This should all be resolved as well 👍